python/lsst/multiprofit/fitting/fit_source.py

```diff
@@ -0,0 +1,897 @@
+                )
+                if compute_errors:
+                    columns_params_radec_err.append(
+                        (
+                            f"{key_base}cen_ra_err",
+                            f"{key_base}cen_dec_err",
+                            f"{key_base}cen_x",
+                            f"{key_base}cen_y",
+                            f"{key_base}cen_x_err",
+                            f"{key_base}cen_y_err",
+                        )
+                    )
+
+        columns = config.schema([channel.name for channel in channels.values()])
+        keys = [column.key for column in columns]
+        idx_flag_first = keys.index("unknown_flag")
+        idx_flag_last = (
+            next(iter(idx for idx, key in enumerate(keys[idx_flag_first:]) if key.endswith("cen_x")))
+            + idx_flag_first
+        )
+        assert idx_flag_last > idx_flag_first
+
+        n_rows = len(catalog_multi)
+        dtypes = [(f'{prefix if col.key != config.column_id else ""}{col.key}', col.dtype) for col in columns]
+        meta = {"config": config.toDict()}
+        results = Table(
+            data=np.full(n_rows, 0, dtype=dtypes),
+            units=[x.unit for x in columns],
+            meta=meta,
+        )
+        self.copy_centroid_errors(
+            columns_cenx_err_copy=columns_cenx_err_copy,
+            columns_ceny_err_copy=columns_ceny_err_copy,
+            results=results,
+            catalog_multi=catalog_multi,
+            catexps=catexps,
+            config_data=config_data,
+        )
+
+        # Validate that the columns are in the right order
+        # assert because this is a logic error if it fails
+        for idx in range(idx_flag_first, idx_flag_last):
+            column = columns[idx]
+            dtype = results.columns[idx].dtype
+            if idx < idx_flag_last:
+                assert dtype == bool
+                assert column.key.endswith("_flag")
+            else:
+                assert dtype == float
+
+        # dummy size for first iteration
+        size, size_new = 0, 0
+        fitInputs = FitInputsDummy()
+        plot = False
+
+        kwargs_err_default = {
+            True: {
+                "options": g2f.HessianOptions(findiff_add=1e-3, findiff_frac=1e-3),
+                "use_diag_only": config.compute_errors_no_covar,
+            },
+            False: {"options": g2f.HessianOptions(findiff_add=1e-6, findiff_frac=1e-6)},
+        }
+
+        range_idx = range(n_rows)
+
+        # TODO: Do this check with dummy data
+        # data, psf_models = config.make_model_data(
+        #     idx_row=range_idx[0], catexps=catexps)
+        # model = g2f.ModelD(data=data, psfmodels=psf_models,
+        #     sources=model_sources, priors=priors)
+        # Remember to filter out fixed centroids from params
+        # assert list(params.values()) == get_params_uniq(model, fixed=False)
```

Comment:
Yeah... the idea is to check if something is badly configured enough that the entire loop is doomed to fail. I used to do this check with the first row only, but as I said, just one row failing isn't enough to predict if the rest will.

This would be nice to have but it's not critical so I've let it slide.